### PR TITLE
Add support for MacroCat Keyboard

### PR DESCRIPTION
**Pull request by starcatmeow**
## Description

Add support for [MacroCat Keyboard](https://github.com/catmunch/macrocat)

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/macrocat/config.h
Comment (zvecr) [inline keyboards/macrocat/config.h:16]:
```suggestion
```

File: keyboards/macrocat/keymaps/default/keymap.c
Comment (zvecr) [inline keyboards/macrocat/keymaps/default/keymap.c:30]:
```suggestion
    }
```

File: keyboards/macrocat/macrocat.h
Comment (zvecr) [inline keyboards/macrocat/macrocat.h:1]: This file should be deleted

File: keyboards/macrocat/rules.mk
Comment (zvecr) [inline keyboards/macrocat/rules.mk:13]:
```suggestion
# This file intentionally left blank
```

File: keyboards/macrocat/info.json
Comment (zvecr) [inline keyboards/macrocat/info.json:13]:
```suggestion
{
    "manufacturer": "Catmunch",
    "keyboard_name": "MacroCat Keyboard",
    "maintainer": "Catmunch",
    "bootloader": "atmel-dfu",
    "diode_direction": "COL2ROW",
    "features": {
        "bootmagic": true,
        "command": false,
        "console": false,
        "extrakey": true,
        "mousekey": true,
        "nkro": true,
        "encoder": true
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/macrocat/keymaps/default/keymap.c
Comment (starcatmeow) [reply keyboards/macrocat/keymaps/default/keymap.c:30]: Thanks for your review, but in our design these two layers are for letting users put their own macros. Deleting these will cause some storage problems of layer 2&3 in VIA.

Comment (zvecr) [reply keyboards/macrocat/keymaps/default/keymap.c:30]: This is not a concern of a default keymap, there should be a via keymap.

File: keyboards/macrocat/keymaps/default/rules.mk
Comment (zvecr) [inline keyboards/macrocat/keymaps/default/rules.mk]: This file should be removed

File: keyboards/macrocat/keymaps/oled/rules.mk
Comment (zvecr) [inline keyboards/macrocat/keymaps/oled/rules.mk:3]:
```suggestion
```

File: keyboards/macrocat/macrocat.c
Comment (zvecr) [inline keyboards/macrocat/macrocat.c:126]:
```suggestion
    matrix_init_user();

    setPinInputHigh(ENCODER_SWITCH);
```

File: keyboards/macrocat/keymaps/default/keymap.c
Comment (starcatmeow) [reply keyboards/macrocat/keymaps/default/keymap.c]: Okay, I created a separate via keymap.

File: keyboards/macrocat/macrocat.c
Comment (zvecr) [inline keyboards/macrocat/macrocat.c:72]:
```suggestion
bool oled_task_kb(void) {
    if (!oled_task_user()) {
        return false;
    }

```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/macrocat/keymaps/via/rules.mk
Comment (zvecr) [inline keyboards/macrocat/keymaps/via/rules.mk:3]:
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]
`keyboards/macrocat/keymaps/default/rules.mk` should be deleted and not an empty file

**Comment by starcatmeow** [comment]
> `keyboards/macrocat/keymaps/default/rules.mk` should be deleted and not an empty file

Thanks. Deleted.

**Comment by starcatmeow** [comment]
> `keyboards/macrocat/keymaps/default/rules.mk` should be deleted and not an empty file

Hi @zvecr , is there anything missing now?

**Comment by starcatmeow** [comment]
Hi @zvecr, I can't find how to mark the change requested as resolved, but I've already deleted that file. Can this PR get merged?

**Comment by zvecr** [comment]
> Pull Request reviews are something that is done in our free time. We are not paid nor compensated for the time we spend reviewing, as it is a labor of love. As such, this means that it can take time for us to get to your Pull Request. Things like family, or life can get in the way of us getting to PRs, and burnout is a serious consideration. The QMK firmware repository averages 200 PRs opened and 200 PRs merged every month, so please have patience.

PRs need 2 approvals before they can be merged. Opening and closing new PRs is only going to make the process take longer.

**Review by drashna — APPROVED** [review]

File: keyboards/macrocat/readme.md
Comment (drashna) [inline keyboards/macrocat/readme.md:3]:
```suggestion
![macrocat](https://i.imgur.com/ItLHGBfh.jpg)
```

Comment (starcatmeow) [reply keyboards/macrocat/readme.md:3]: Thanks for your advice, but what does the extra h do?

File: keyboards/macrocat/keymaps/default/keymap.c
Comment (zvecr) [inline keyboards/macrocat/keymaps/default/keymap.c:29]:
```suggestion
```

**Review by zvecr — APPROVED** [review]

**Pull request merged**
